### PR TITLE
feat: add oracles v2 endpoint without making use of streams

### PR DESCRIPTION
**Pull request by sborrazas**
Output from benchmarks (pretty similar to active/inactive output):
```
Name                 ips        average  deviation         median         99th %
oracles_v2         19.89       50.27 ms     ±8.09%       49.80 ms       74.77 ms
oracles            18.40       54.35 ms     ±9.09%       53.14 ms       71.95 ms

Comparison: 
oracles_v2         19.89
oracles            18.40 - 1.08x slower +4.09 ms

Memory usage statistics:

Name               average  deviation         median         99th %
oracles_v2       672.86 KB     ±0.10%      672.62 KB      674.34 KB
oracles          767.22 KB     ±0.09%      767.46 KB      767.57 KB

Comparison: 
oracles_v2       672.62 KB
oracles          767.22 KB - 1.14x memory usage +94.36 KB
```

This is the commit that needs to be checked out to run these benchmarks f369fb662f3e60949c52005b9a51af4b00d7592b

### CODE REVIEW

**Review by thepiwo — COMMENTED** [review]
Is there any downside or incompatibility to not using streams, if not, why make a v2?

**Comment by sborrazas** [comment]
@thepiwo there isn't really any, I tested them and they return the same thing. Should I just get rid of v2 and merge it into the non-v2 routes instead?

**Comment by thepiwo** [comment]
@sborrazas if there is no incompatibility lets make non-stream the new default

File: lib/ae_mdw/oracles.ex [hunk 1]
```diff
@@ -7,11 +7,12 @@ defmodule AeMdw.Oracles do
 
   alias AeMdw.Db.Format
   alias AeMdw.Db.Model
+  alias AeMdw.Db.Util
   alias AeMdw.Mnesia
   alias AeMdw.Node
 
   @type cursor :: binary()
-  # TODO: This needs to be an actual type like AeMdw.Db.Oracle.t()
+  # TO_DO: This needs to be an actual type like AeMdw.Db.Oracle.t()
```
Comment (jyeshe) [inline lib/ae_mdw/oracles.ex:15]: formatter? (:

File: lib/ae_mdw_web/plugs/paginated_plug.ex [hunk 2]
```diff
@@ -21,25 +22,45 @@ defmodule AeMdwWeb.Plugs.PaginatedPlug do
...
-        {limit, ""} when limit <= @max_limit and limit > 0 -> limit
-        {_limit, _rest} -> @default_limit
-        :error -> @default_limit
-      end
-
-    conn
-    |> assign(:direction, direction)
-    |> assign(:cursor, Map.get(params, "cursor"))
-    |> assign(:limit, limit)
+    with {:ok, direction} <- extract_direction(params),
```
Comment (jyeshe) [inline lib/ae_mdw_web/plugs/paginated_plug.ex:25]: :clap:

**Review by jyeshe — COMMENTED** [review]
I think we could have intermediary tests to validate the replacement. Some commit before final where the integration tests of the new v1 controller is run against the old v1 (likewise we had v2 and v1).

**Review by thepiwo — APPROVED** [review]
looks good if comments from @jyeshe are resolved

**Review by jyeshe — APPROVED** [review]
Looking good and good point on the tests, we can focus on the middleware state. For the validation of transitions, we count on the help of experienced guys within testnet.

File: lib/ae_mdw/mnesia.ex [hunk 3]
```diff
@@ -68,6 +84,7 @@ defmodule AeMdw.Mnesia do
       :mnesia.async_dirty(fn ->
         case :mnesia.read(tab, first_key) do
           @end_token -> {[], nil}
```
[the remark this answers is not in the document]
Comment (sborrazas) [reply lib/ae_mdw/mnesia.ex]: Done, thanks for catching it!

File: bench/ae_mdw_web/oracle_controller_bench.exs [hunk 4]
```diff
@@ -8,34 +8,39 @@ defmodule AeMdw.OracleControllerBench do
...
     build_conn()
-    |> get("/oracles/inactive?direction=forward")
+    |> get("/oracles")
   end
 
-  def inactive_oracles_v2 do
+  def oracles_v2 do
     build_conn()
-    |> get("/v2/oracles/inactive/forward")
+    |> get("/v2/oracles")
```
[the remark this answers is not in the document]
Comment (sborrazas) [reply bench/ae_mdw_web/oracle_controller_bench.exs]: The tests are pretty much the same, and they all pass without any modifications

**Pull request merged**